src/VecSim/query_results.h

```diff
@@ -31,7 +33,10 @@ float VecSimQueryResult_GetScore(const VecSimQueryResult *item);
 /**
  * @brief An opaque object from which results can be obtained via iterator.
  */
-typedef VecSimQueryResult *VecSimQueryResult_List;
+typedef struct {
+    VecSimQueryResult *results;
+    VecSimQueryResult_Code code;
+} VecSimQueryResult_List;
```

Comment:
let's consider renaming it. It is not a list anymore

Reply:
A list with a header is a list in my opinion. how will you call it?